lib/public_xml_record.rb

```diff
@@ -31,9 +33,21 @@ def catkey
 
   # @return objectLabel value from the DOR identity_metadata, or nil if there is no barcode
   def label
+    return nil unless collection_in_searchworks?
+
     get_value(public_xml_doc.xpath('/publicObject/identityMetadata/objectLabel'))
   end
 
+  def collection_in_searchworks?
```

Comment:
I think I've been tripped up by this method name and/or the context for how this gets used. Maybe it's just the  method name, but I wonder if there's a more explicit/clearer way to implement this (kind of confusing in the first place) requirement.

From your description, it sounds like we're trying to suppress the id and title from the `collection` and `collection_with_title` fields?

https://github.com/sul-dlss/searchworks_traject_indexer/blob/main/lib/traject/config/sdr_config.rb#L295-L303

Instead of tweaking the underlying data for collection objects (which.. feels like it might be useful in some contexts.. debugging if nothing else?), what if `PublicXmlRecord` had a `released_collections` accessor that we used anywhere we needed it?

Reply:
That works. The upside of this is we don't have to make any changes to the code in searchworks or the solr index. I have also left collection_with_title to still index with this it just returns an empty value of -|- which I thought would be good for debugging. Basically it would indicate that this has a collection attached but it isn't public. Probably not the best indicator so I can update the code, but I think that means we would need to update the solr index along with Searchworks.